Framework/HistogramData/src/HistogramMath.cpp

```diff
@@ -93,6 +98,55 @@ Histogram &operator-=(Histogram &histogram, const Histogram &other) {
   return histogram;
 }
 
+/// Multiplies data in other Histogram with this Histogram, propagating
+/// uncertainties.
+Histogram &operator*=(Histogram &histogram, const Histogram &other) {
+  if (histogram.yMode() == Histogram::YMode::Counts &&
+      other.yMode() == Histogram::YMode::Counts)
+    throw std::runtime_error("Invalid operation: Cannot multiply two "
+                             "histograms with YMode::Counts. Convert (one of "
+                             "them) to a distribution (YMode::Frequencies).");
+  checkSameXMode(histogram, other);
+  checkSameX(histogram, other);
+
+  auto &y1 = histogram.mutableY();
+  auto &e1 = histogram.mutableE();
+  const auto &y2 = other.y();
+  const auto &e2 = other.e();
+  for (size_t i = 0; i < y1.size(); ++i)
+    e1[i] = sqrt(pow(e1[i] * y2[i], 2) + pow(e2[i] * y1[i], 2));
```

Comment:
Verified that this formula matches that in [Multiply.cpp](https://github.com/mantidproject/mantid/blob/0deb3d033a3699dcdcb9b1e548fd1f945f79356d/Framework/Algorithms/src/Multiply.cpp#L32)